### DISCUSSION TITLE
open a view and close another: ActionHandler

### DISCUSSION THREAD

**Discussion by Adem-Kri** [eclipse-theia/theia#12037] [General] [answered]
Hello,
I created a button inside my glsp view similar to the button overlay showcase existed in the documentation.
So, I have a glsp view  and a button and I want to implement an actionHandler that closes the opened view and open a theia extension,
this is the button source code, inside `Switch()` function is where the process will be implemented
```
containerElement.appendChild(
        this.createButton(
          "btn_switch_diagram",
          "Switch to table view",
          "arrow-both",
          SwitchAction.Switch()
        )
      );
```
I have made some researches and came out with this: I need to implement the `openerService `and `widgetOpenerOptions`,
but I am still a beginner with theia and didn't know how to do it, am seeking for your help and thank you in advance.

**Comment by planger** [comment, marked as answer]
Just for references, this question has been answered in the GLSP discussions: https://github.com/eclipse-glsp/glsp/discussions/838
